numba/core/types/abstract.py

```diff
@@ -89,19 +91,22 @@ class Type(metaclass=_TypeMetaclass):
     # Rather the type is reflected at the python<->nopython boundary
     reflected = False
 
-    def __init__(self, name):
+    # Set in _TypeMetaclass call method.
+    _code: int
+
+    def __init__(self, name: str):
         self.name = name
 
     @property
-    def key(self):
+    def key(self) -> pt.Any:
         """
         A property used for __eq__, __ne__ and __hash__.  Can be overridden
         in subclasses.
         """
         return self.name
 
     @property
-    def mangling_args(self):
+    def mangling_args(self) -> pt.Tuple[str, pt.Tuple[pt.Any, ...]]:
```

Comment:
the docstring says that a sequence is returned so I'd write it as `def mangling_args(self) -> pt.Tuple[str, pt.Sequence[pt.Any]]:`
In fact, there's one case where a list is returned (not a string).

Reply:
Do you mean "(not a tuple)"?  I updated the annotation based on that interpretation.

Reply:
sorry, I meant "not a tuple".